osu.Game/OsuGame.cs

```diff
@@ -426,9 +426,8 @@ public void PresentScore(ScoreInfo score, ScorePresentType presentType = ScorePr
         {
             // The given ScoreInfo may have missing properties if it was retrieved from online data. Re-retrieve it from the database
             // to ensure all the required data for presenting a replay are present.
-            var databasedScoreInfo = score.OnlineScoreID != null
-                ? ScoreManager.Query(s => s.OnlineScoreID == score.OnlineScoreID)
-                : ScoreManager.Query(s => s.Hash == score.Hash);
+            // Todo: This should use the OnlineScoreID if available, however lazer scores are imported without an OnlineScoreID for the time being (see Player.ImportScore()).
```

Comment:
If lazer scores don't have an `OnlineScoreID` shouldn't it have used the `Hash` equality in the previous code? Not quite seeing how this was failing from the diff alone..

Is it the case that they have an `OnlineScoreID` in the `Present` call but not in the database?

Reply:
Because they're given an OnlineScoreId after the import, in order to dedupe for the results screen.

Reply:
I see. And currently the present method is only used for lazer scores to it will work correctly or something? Just seeing cases where hash is arbitrarily set like

![20210615 114058 (JetBrains Rider-EAP)](https://user-images.githubusercontent.com/191335/121984801-957ebc00-cdce-11eb-8af3-b1bd001aed9c.png)

makes me a bit wary..

Reply:
You may be on to something with that.

Reply:
Turns out this did indeed break online scores. Changed it to use hash as the fallback and tested watching replays from:
- Local score.
- Online score.
- Played score from results screen.